node/plugin/tests/plugin_test.go

```diff
@@ -266,3 +317,49 @@ func getTransactor(t *testing.T, operator deploy.OperatorVars) *eth.Writer {
 
 	return transactor
 }
+
+// runNodePlugin runs the node plugin directly using go run
+func runNodePlugin(t *testing.T, operation string, operator OperatorConfig) {
+	t.Helper()
+
+	ctx := t.Context()
+	socket := string(core.MakeOperatorSocket(
+		operator.NODE_HOSTNAME,
+		operator.NODE_DISPERSAL_PORT,
+		operator.NODE_RETRIEVAL_PORT,
+		operator.NODE_V2_DISPERSAL_PORT,
+		operator.NODE_V2_RETRIEVAL_PORT,
+	))
+
+	// Get the path to the node plugin cmd directory relative to this file
+	_, filename, _, _ := runtime.Caller(0)
+	testDir := filepath.Dir(filename)
+	rootDir := filepath.Join(testDir, "..", "..", "..")
+	pluginCmdPath := filepath.Join(rootDir, "node", "plugin", "cmd")
+
+	// Run the plugin directly with go run
+	cmd := exec.CommandContext(ctx, "go", "run", pluginCmdPath,
+		"--operation", operation,
+		"--ecdsa-key-file", operator.NODE_ECDSA_KEY_FILE,
+		"--bls-key-file", operator.NODE_BLS_KEY_FILE,
+		"--ecdsa-key-password", operator.NODE_ECDSA_KEY_PASSWORD,
+		"--bls-key-password", operator.NODE_BLS_KEY_PASSWORD,
+		"--socket", socket,
+		"--quorum-id-list", operator.NODE_QUORUM_ID_LIST,
+		"--chain-rpc", operator.NODE_CHAIN_RPC,
+		"--eigenda-directory", operator.NODE_EIGENDA_DIRECTORY,
+		"--bls-operator-state-retriever", operator.NODE_BLS_OPERATOR_STATE_RETRIVER,
+		"--eigenda-service-manager", operator.NODE_EIGENDA_SERVICE_MANAGER,
+		"--churner-url", operator.NODE_CHURNER_URL,
+		"--num-confirmations", "0",
```

Comment:
what exactly are we trying to test with this? that the flags work properly? If not then why don't we just import the plugin function and call it directly, instead of subshelling like this? Do we gain much from doing this this way?

Reply:
Going to leave it as is, the motivation from the original test writer was to test that the CLI functions properly.